lib/internal/fs/watch/linux.js

```diff
@@ -23,20 +22,25 @@ function lazyLoadFsSync() {
   return internalSync;
 }
 
-function traverse(dir, files = new SafeMap()) {
-  const { readdirSync } = lazyLoadFsSync();
+async function traverse(dir, parent = undefined, files = new SafeMap()) {
+  const { readdir, stat } = lazyLoadFsPromises();
 
-  const filenames = readdirSync(dir, { withFileTypes: true });
+  const filenames = await readdir(dir, { withFileTypes: true });
 
-  files.set(dir, new Dirent(dir, UV_DIRENT_DIR));
+  if (parent !== undefined) {
+    files.set(dir, parent);
```

Comment:
Is this necessary? Wouldn't the call line 40 have already set it?

Reply:
Thanks! You are right. I removed it.